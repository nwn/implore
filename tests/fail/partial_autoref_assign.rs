use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op(autoref)]
fn sub_assign(_lhs: &mut X, _rhs: X) { }

fn partial_subtraction() {
    let mut x = X;
    assert_eq!((), x -= X);
    assert_eq!((), x -= &X);
}

fn main() {}
