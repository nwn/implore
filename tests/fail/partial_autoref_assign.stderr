error[E0308]: mismatched types
  --> $DIR/partial_autoref_assign.rs:12:25
   |
12 |     assert_eq!((), x -= &X);
   |                         ^^
   |                         |
   |                         expected struct `X`, found `&X`
   |                         help: consider removing the borrow: `X`
